### PR TITLE
Pattern Matcher Tests

### PR DESCRIPTION
**Pull request by reddyn12**
This is a reopen of #4744 because I'm confident that the closing comment was for another pr as this pr only dealt with testing the existing pattern matcher functionality. There was no line getbacks or functionality changes. Once this pr gets merged to master, I can be able to prove that #4670 introduces no functional changes.

### CODE REVIEW

File: test/test_pattern_matcher.py [hunk 1]
```diff
@@ -127,6 +129,117 @@ def test_allow_len(self):
...
+
+  def test_neg_cmplt(self):
+    matcher = PatternMatcher([({"uop": UOps.ALU, "arg": BinaryOps.CMPLT, "vin": ({"uop": UOps.ALU, "arg": UnaryOps.NEG, "vin": ({"__name__": "x"},)},
+                        {"__name__": "c", "uop": UOps.CONST, "dtype": dtypes.int})},
+                        lambda c,x: UOp(UOps.ALU, dtypes.bool, (UOp.const(c.dtype, -c.arg), x), BinaryOps.CMPLT))])
+    c1 = UOp(UOps.CONST, dtypes.int, arg=42)
+    c1_neg = UOp(UOps.CONST, dtypes.int, arg=-42)
+    c2 = UOp(UOps.CONST, dtypes.int, arg=50)
+    a1 = UOp(UOps.ALU, dtypes.int, (c2,), UnaryOps.NEG)
+    a2 = UOp(UOps.ALU, dtypes.int, (a1,c1), BinaryOps.CMPLT)
```
Comment (Qazalin) [inline test/test_pattern_matcher.py:239]: err, CMPLT is always bool

File: test/test_pattern_matcher.py [hunk 2]
```diff
@@ -127,6 +129,117 @@ def test_allow_len(self):
     self.assertEqual(matcher.rewrite(c6), None)
     self.assertEqual(matcher.rewrite(c7), c7)
 
+  def test_constant_arg(self):
```
Comment (Qazalin) [inline test/test_pattern_matcher.py:132]: redundant test, we already have `test_arg`

Comment (reddyn12) [reply test/test_pattern_matcher.py]: removed test

**Comment by chenyuxyz** [comment]
i don't like this, it's pretty ugly and i will be sad to maintain this. like what is `test_max_214` and what's this formatting?

**Pull request closed without merging**